src/main/java/pl/pelikan/service/UserService.java

```diff
@@ -0,0 +1,80 @@
+package pl.pelikan.service;
+
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.pelikan.model.User;
+import pl.pelikan.model.UserRole;
+import pl.pelikan.repository.UserRepository;
+import pl.pelikan.repository.UserRoleRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class UserService {
+
+    private static final String DEFAULT_ROLE = "ROLE_USER";
+    private final UserRepository userRepository;
+    private final UserRoleRepository roleRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public UserService(UserRepository userRepository, UserRoleRepository roleRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void registerUser(User user) {
+        UserRole defaultRole = roleRepository.findByRole(DEFAULT_ROLE);
+        user.getRoles().add(defaultRole);
+        String passwordHash = passwordEncoder.encode(user.getPassword());
+        user.setPassword(passwordHash);
+        userRepository.save(user);
+    }
+
+    public User findByUsername(String username) {
+        return userRepository.findByUsername(username).get();
+    }
+
+    public void updateUser(User user) {
+        userRepository.save(user);
+    }
+
+    public void changeCredentials(User user) {
+        User toUpdate = findCurrentUser();
+        toUpdate.setFirstName(user.getFirstName());
+        toUpdate.setLastName(user.getLastName());
+        toUpdate.setPassword(passwordEncoder.encode(user.getPassword()));
+        updateUser(toUpdate);
+    }
+
+    public void changeUserRole(Long id) {
```

Comment:
Z doświadczenia powiedziałbym, że lepiej przekazać już z frontendu o jaką akcję nam chodzi - czy to dodać, czy usunąć daną rolę, a nie robić tzw. "tick". Jak ktoś przez przypadek zrobi double click (bez refresha) - pójdą dwa requesty i nic się nie zmieni. A jakbyśmy wysłali, że chodziło przykładowo o usunięcie tej roli już z fronta, to wszystko dalej zadziała poprawnie